File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        upper = max(piles) * len(piles)
        lower = 1
        rate = upper
        while lower <= upper:
            mid = lower + (upper - lower) // 2
            totalTime = 0
            for p in piles:
                totalTime += int(math.ceil(float(p) / mid))
            if totalTime > h:
                lower = mid + 1
            else:
                rate = min(rate, mid)
                upper = mid - 1
        return rate

File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() < 2){
            return nums[0];
        }
        vector<int> dp = nums;
        for(int i = 2; i < nums.size(); i++){
            dp[i] = dp[i-2] + dp[i];
        }

        return max(dp[dp.size()-1], dp[dp.size()-2]);
    }
};
